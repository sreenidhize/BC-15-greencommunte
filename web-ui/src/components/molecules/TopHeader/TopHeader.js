import React from 'react';
import PropTypes from 'prop-types';
import { Location } from '../../atoms/Location/Location';
import { Avatar, makeStyles } from "@material-ui/core";
import Grid from '@material-ui/core/Grid';
import MessageOutlinedIcon from '@material-ui/icons/MessageOutlined';
import NotificationsNoneOutlinedIcon from '@material-ui/icons/NotificationsNoneOutlined';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
    root: {
        color: "#9bbdcb",
        width: "20px",
        height: "20px",
        margin: "10px",
    },
    right: {
        textAlign: "right",
        paddingRight: "100px",
    },
    label: {
        fontSize: "16px",
        paddingLeft: "10px",
        position: "relative",
        top: "10px",
    },
    avatarMargin: {
        marginLeft: "10px",
    },
    location: {
        paddingLeft: '80px',
    }
});

const TopHeader = ({ locationLabel, profileName }) => {
    
    const classes = useStyles();

    return (

        <Grid container alignItems="center" justify="space-between" data-testid="topheader-placeholder" >
            <Grid item xs={3} className={classes.location} >
                <Location label={locationLabel||"Hyderabad, Telangana, India"} />
            </Grid>

            <Grid item container xs={3} >
                <MessageOutlinedIcon className={classes.root} />
                <NotificationsNoneOutlinedIcon className={classes.root} />
                <Avatar className={classes.avatarMargin} src={'/assets/avatar_profile.jpeg'}>{profileName.charAt(0)}</Avatar>
                <Typography variant="h4" className={classes.label} >{profileName}</Typography>
                <ExpandMoreIcon className={classes.root} />
            </Grid>
        </Grid>
    );
};

TopHeader.propTypes = {
    locationLabel: PropTypes.string,
    profileName: PropTypes.string,
};

export default TopHeader;