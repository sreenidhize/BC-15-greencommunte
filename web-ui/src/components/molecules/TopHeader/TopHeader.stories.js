import React from 'react';
import PropTypes from 'prop-types';
import TopHeader from './TopHeader';
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';

export default {
    title: 'molecules/TopHeader',
    component: TopHeader,
};

const Template = (args) => <ThemeProvider theme={theme}><TopHeader {...args} /></ThemeProvider>;

export const TopHeaderOne = Template.bind({});
TopHeaderOne.args = {
    locationLabel: 'East Marredpally, Secundrabad',
    profileName: 'Sara Joseph',
};

export const TopHeaderTwo = Template.bind({});
TopHeaderTwo.args = {
    locationLabel: 'East Marredpally, Hyderabad',
    profileName: 'Joseph Sara',
};