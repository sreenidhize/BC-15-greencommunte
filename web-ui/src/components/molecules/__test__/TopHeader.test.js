import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import TopHeader from '../TopHeader/TopHeader';

describe("TopHeader placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <TopHeader locationLabel="hyderabad" profileName="Bob" />
    );
    expect(container).toMatchSnapshot();
  });

  test("test TopHeader render", () => {
    const { getByTestId } = render(<TopHeader  locationLabel="hyderabad" profileName="Bob"/>);
    expect(getByTestId("topheader-placeholder")).toBeInTheDocument();
  });
});