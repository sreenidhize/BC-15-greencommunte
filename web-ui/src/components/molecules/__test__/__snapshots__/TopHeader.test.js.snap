// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeader placeholder test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    data-testid="topheader-placeholder"
  >
    <div
      class="MuiGrid-root makeStyles-location-5 MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        data-testid="location-placeholder"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-root-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
          />
          <circle
            cx="12"
            cy="9"
            r="2.5"
          />
        </svg>
        <h4
          class="MuiTypography-root makeStyles-label-7 MuiTypography-h4"
        >
          hyderabad
        </h4>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 4h16v12H5.17L4 17.17V4m0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2H4zm2 10h12v2H6v-2zm0-3h12v2H6V9zm0-3h12v2H6V6z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
        />
      </svg>
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarMargin-4"
      >
        <img
          class="MuiAvatar-img"
          src="/assets/avatar_profile.jpeg"
        />
      </div>
      <h4
        class="MuiTypography-root makeStyles-label-3 MuiTypography-h4"
      >
        Bob
      </h4>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
